### PR TITLE
Solidstart/followers following count

### PR DESCRIPTION
**Pull request by AllStackDev1**
Closes: #1475

image

# Description
- Followers and Following counts not displaying correct values - always zero

Image

### CODE REVIEW

File: solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx [hunk 1]
```diff
@@ -9,103 +9,100 @@ import { Icon } from 'solid-heroicons';
 import { TwitterIcon } from '../Icons';
 import OrgList from './OrgList';
 import styles from './UserProfile.module.css';
-import { Show } from 'solid-js';
+import { Show, splitProps } from 'solid-js';
+import { UserProfileProps } from '~/types/user-profile-type';
```
Comment (BigAB) [inline solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx:13]: Why would you move the props for this component to a separate file?

I think the props type should be colocated  with the component.

File: solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx [hunk 2]
```diff
@@ -9,103 +9,100 @@ import { Icon } from 'solid-heroicons';
...
-  organizations: {
-    nodes: {
-      avatarUrl: string;
-      name: string;
-      login: string;
-    }[];
-  };
-}
+const UserProfile = (props: UserProfileProps) => {
+  const [local] = splitProps(props, [
```
Comment (BigAB) [inline solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx:16]: Is splitting here needed? It looks like you take ever property in the `local`

File: solidstart-tanstackquery-tailwind-modules/src/services/get-user-profile.ts [hunk 3]
```diff
@@ -2,15 +2,15 @@ import FetchApi, { ApiProps } from './api';
...
+import { UserProfileProps } from '~/types/user-profile-type';
 
 type UserProfileVariables = {
   username: string;
 };
 
 type Response = {
   data: {
-    user: UserProfile;
+    user: UserProfileProps;
```
Comment (BigAB) [inline solidstart-tanstackquery-tailwind-modules/src/services/get-user-profile.ts:13]: So in general, the props of a component, or params of a function call, would (or can) be a different type than the entity type itself.

The `user` should probably be either type `User` or maybe `Profile` or `UserProfile` but it being `UserProfileProps` doesn't seem right.

If you had a `UserProfile` type, and the `Profile` components PropsType happened to exactly match the `UserProfile` type, than I would possible just have the Props be type as a `UserProfile` but the other way around seems backwards

File: solidstart-tanstackquery-tailwind-modules/src/types/user-profile-type.ts [hunk 4]
```diff
@@ -1,21 +1,21 @@
-export interface UserProfile {
+export interface UserProfileProps {
```
Comment (BigAB) [inline solidstart-tanstackquery-tailwind-modules/src/types/user-profile-type.ts:1]: As noted above, this change does not seem like a good idea.

I would keep the component props type separate, and maybe even be a subset or extension of this `UserProfile` type, but don't put the props type here

**Review by BigAB — CHANGES_REQUESTED** [review]
These comments are all pretty much related to the same thing.

File: solidstart-tanstackquery-tailwind-modules/src/services/get-user-profile.ts (hunk 3 above)
Comment (AllStackDev1) [reply solidstart-tanstackquery-tailwind-modules/src/services/get-user-profile.ts:13]: Sure will update it.

File: solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx (hunk 2 above)
Comment (AllStackDev1) [reply solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx:16]: Not really. updating it.

Comment (AllStackDev1) [reply solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx:16]: Splitting actually isn't needed here.

File: solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx (hunk 1 above)
Comment (AllStackDev1) [reply solidstart-tanstackquery-tailwind-modules/src/components/UserProfile/index.tsx:13]: The props is been used in other places.

File: solidstart-tanstackquery-tailwind-modules/src/types/user-profile-type.ts (hunk 4 above)
Comment (AllStackDev1) [reply solidstart-tanstackquery-tailwind-modules/src/types/user-profile-type.ts:1]: So this type also matches the API result we get from the query, and fits precisely with the props the component requires, that this why it is declared in the types/ to be used in places it is needed.

**Review by BigAB — APPROVED** [review]
I think this is good

**Pull request merged**